web/client/src/library/components/graph/help.ts

```diff
@@ -346,30 +313,85 @@ function toNodeOrEdgeId(...args: Array<string | undefined>): string {
   return args.filter(Boolean).join('__')
 }
 
-function toggleLineageColumn(
-  action: 'add' | 'remove',
-  modelName: string,
-  columnName: string,
-  connections: { ins: string[]; outs: string[] } = { ins: [], outs: [] },
-  removeActiveEdges: (edges: string[]) => void,
-  addActiveEdges: (edges: string[]) => void,
-): void {
-  const columnId = toNodeOrEdgeId(modelName, columnName)
-  const sourceId = toNodeOrEdgeId('source', columnId)
-  const targetId = toNodeOrEdgeId('target', columnId)
-
-  const edges = [
-    connections.ins.map(id => toNodeOrEdgeId('source', id)),
-    connections.outs.map(id => toNodeOrEdgeId('target', id)),
-  ]
-    .flat()
-    .concat([sourceId, targetId])
-
-  if (action === 'remove') {
-    removeActiveEdges(edges)
+// TODO: use better merge
```

Comment:
What's the plan for this function? You mentioned you tested lineage with like 1k models, each with hundreds of columns. Is this a bottleneck with its triply nested for loops?